src/storage-types/src/instances.rs

```diff
@@ -25,13 +25,13 @@ include!(concat!(env!("OUT_DIR"), "/mz_storage_types.instances.rs"));
 )]
 pub enum StorageInstanceId {
     /// A system storage instance.
-    System(u64),
+    System(u32),
     /// A user storage instance.
-    User(u64),
+    User(u32),
```

Comment:
I like the `u40`/`u48` idea. I'll try and whip that up. You would probably know the answer to this better than me, is 255 enough for all the introspection sources we ever might have?

Reply:
Also if we see people start to get close to `u40`/`u48`, we could either:

- Go back to a `u64` and increase the size of `GlobalId`.
- Convert `GlobalId` to a `u128` and manually pack it ourselves. That's more than enough space to store all the information we need, even with a `Cluster: u64`.

Reply:
> is 255 enough for all the introspection sources we ever might have?

Currently we have ~30, and every source has some maintenance and implementation overhead, so at least with the current compute logging system I doubt that we'd ever get more than 255 sources. If I'm wrong we still have the option to steal a couple bit from the 8 bits we reserve for the GlobalId variant, so that wouldn't be the end of the world.

> Also if we see people start to get close to u40/u48, we could either:

Agreed! The challenge might be noticing that people get close :)

Reply:
> Agreed! The challenge might be noticing that people get close :)

I'll add an error log when someone fills up 47 bits so we are alerted.

Reply:
Done, I just increased this to 48 bits.